packages/Search/src/details/DTK_DetailsTreeTraversal.hpp

```diff
@@ -177,22 +178,25 @@ nearestQuery( BoundingVolumeHierarchy<DeviceType> const &bvh,
         double const node_distance = queue.top().second;
         // NOTE: it would be nice to be able to do something like
         // tie( node, node_distance = queue.top();
-        queue.pop();
+
+        // NOTE: not calling queue.pop() here so that it can be combined with
+        // the next push in case the node is internal (thus sparing a bubble-up
+        // operation)
         if ( TreeTraversal<DeviceType>::isLeaf( node ) )
         {
+            queue.pop();
             insert( TreeTraversal<DeviceType>::getIndex( bvh, node ),
                     node_distance );
             count++;
         }
         else
         {
             // insert children of the node in the priority list
-            for ( Node const *child :
-                  {node->children.first, node->children.second} )
-            {
-                double const child_distance = distance( child );
-                queue.push( child, child_distance );
-            }
+
+            auto const left_child = node->children.first;
+            auto const right_child = node->children.second;
+            queue.pop_push( left_child, distance( left_child ) );
+            queue.push( right_child, distance( right_child ) );
```

Comment:
Can some smart person figure out whether it's generally a better idea to first push left or right child here? Or does it matter?

Reply:
I'll take the bait.
I measured the exact same for CUDA, and 0-3% slower on CPU when pop_pushing the right child first.

Reply:
@hydroo Thanks. It's good to know that it makes no difference.